### ISSUE TITLE
[1.13] Random disconnect when joining singleplayer world

### ISSUE THREAD

**Issue by ErdbeerbaerLP** [MinecraftForge/MinecraftForge#5590]
Not much to say, the game was stuck at loading world 96%

Log output:
```
[20:47:43.169] [Server thread/INFO] [minecraft/MinecraftServer]: Preparing spawn area: 96%
[20:47:43.217] [Server thread/INFO] [minecraft/MinecraftServer]: Preparing spawn area: 100%
[20:47:43.218] [Server thread/INFO] [minecraft/MinecraftServer]: Time elapsed: 995 ms
[20:47:43.266] [Server thread/INFO] [minecraft/IntegratedServer]: Changing view distance to 12, from 10
[20:47:43.537] [Server thread/INFO] [minecraft/NetHandlerLoginServer]: com.mojang.authlib.GameProfile@67b8e4e0[id=dfdc5dbd-4a0e-3b6a-a815-601a03741434,name=Player116,properties={},legacy=false] (local:E:7902947a) lost connection: Internal Exception: java.lang.ClassCastException: net.minecraft.network.NetHandlerLoginServer cannot be cast to net.minecraft.network.play.INetHandlerPlayClient
```

**Comment by mrburger** [comment]
I have been having this issue as well. Originally I thought it was my mod's fault, but it persists across all versions of 1.13.2 forge I have tested. I have listed where it occurs and how to reproduce below, despite the intermittency of the issue. 
### **Tested Scenarios**

- Vanilla worlds only
- Modded worlds only
- Vanilla world then modded world
- 1.13.2 server log-out-in
- Dev environment

### **Reproduction Tips**
The issue is intermittent, and does not always occur. for me, quickly loading and unloading worlds with similar names regardless of type seems to cause it the most reliably. It happened most frequently when I was quickly generating "New World" worlds, immediately leaving and deleting them, and then quickly generating another "New World" world. The issue is most prevalent in the dev environment, but  I have received reports of it happening with my mod in the exact same scenario. I tested it without the mod, and it still happens.

### **Parting Notes**
The issue did not occur in my testing today, but it was a constant annoyance while developing my mod.  I hosted a mod-less forge 1.13.2 server, and upon leaving the server (either by choice or by being kicked), and rejoining, the user would receive a NPE from exactly the same error as defined above.

**Comment by JoJoDeveloping** [comment]
Should have been fixed by d8d30e1f

**Comment by cpw** [comment]
Closing, it should have been fixed by the commit above. Note that there is a vanilla bug here : the exception handling code in NetworkManager assumes the exception is always fired on the server, and only sends a server disconnection packet.
